From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ambrus Toth <palpista11@gmail.com>
Date: Wed, 30 Sep 2026 21:47:29 +0200
Subject: [PATCH 13/13] Build swiftpm-testing-helper with CMake

SwiftPM only builds `swiftpm-testing-helper` on Darwin when it is built
with SwiftPM (`#if canImport(Darwin)` in `Package.swift`), but the
CMake build does not build it at all. `swift test` needs it on Darwin.

See https://github.com/swiftlang/swift-package-manager/issues/10600.
---
 Sources/CMakeLists.txt                        |  5 +++++
 Sources/swiftpm-testing-helper/CMakeLists.txt | 18 ++++++++++++++++++
 2 files changed, 23 insertions(+)
 create mode 100644 Sources/swiftpm-testing-helper/CMakeLists.txt

diff --git a/Sources/CMakeLists.txt b/Sources/CMakeLists.txt
index d4e659a..4a7679c 100644
--- a/Sources/CMakeLists.txt
+++ b/Sources/CMakeLists.txt
@@ -47,6 +47,11 @@ add_subdirectory(swift-package-collection)
 add_subdirectory(swift-package-registry)
 add_subdirectory(swift-run)
 add_subdirectory(swift-test)
+
+if(APPLE)
+  add_subdirectory(swiftpm-testing-helper)
+endif()
+
 add_subdirectory(SwiftSDKCommand)
 add_subdirectory(Workspace)
 add_subdirectory(XCBuildSupport)
diff --git a/Sources/swiftpm-testing-helper/CMakeLists.txt b/Sources/swiftpm-testing-helper/CMakeLists.txt
new file mode 100644
index 0000000..ef791f6
--- /dev/null
+++ b/Sources/swiftpm-testing-helper/CMakeLists.txt
@@ -0,0 +1,18 @@
+# This source file is part of the Swift open source project
+#
+# Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+add_executable(swiftpm-testing-helper
+  Entrypoint.swift)
+set_target_properties(swiftpm-testing-helper PROPERTIES
+  Swift_MODULE_NAME swiftpm_testing_helper)
+
+target_compile_options(swiftpm-testing-helper PRIVATE
+  -parse-as-library)
+
+install(TARGETS swiftpm-testing-helper
+  RUNTIME DESTINATION "${CMAKE_INSTALL_LIBEXECDIR}/swift/pm")
